c2corg_ui/views/document.py

```diff
@@ -413,20 +421,43 @@ def _get_api_cache_key_from_etag(self, etag):
     def _get_response(self, page_html):
         return get_response(self.request, page_html)
 
+    def _redirect(self, id, lang, slug=None):
+        location = ''
+        if slug is None:
+            location = self.request.route_url(
+                self._API_ROUTE + '_view_id_lang', id=id, lang=lang)
+        else:
+            location = self.request.route_url(
+                self._API_ROUTE + '_view', id=id, lang=lang, slug=slug)
+        raise HTTPMovedPermanently(location=location)
```

Comment:
It's better if we use `HTTPFound` (302) instead of `HTTPMovedPermanently` (301). The browser is very eager in caching 301 redirects, it will *never* do a request again and directly use the redirected URL. I will also change it in the API.